src/core/ext/xds/xds_client_stats.cc

```diff
@@ -131,29 +131,43 @@ XdsClusterLocalityStats::~XdsClusterLocalityStats() {
 
 XdsClusterLocalityStats::Snapshot
 XdsClusterLocalityStats::GetSnapshotAndReset() {
-  Snapshot snapshot = {
-      GetAndResetCounter(&total_successful_requests_),
-      // Don't reset total_requests_in_progress because it's
-      // not related to a single reporting interval.
-      total_requests_in_progress_.load(std::memory_order_relaxed),
-      GetAndResetCounter(&total_error_requests_),
-      GetAndResetCounter(&total_issued_requests_),
-      {}};
-  MutexLock lock(&backend_metrics_mu_);
-  snapshot.backend_metrics = std::move(backend_metrics_);
+  Snapshot snapshot;
+  for (auto& percpu_stats : stats_) {
+    Snapshot percpu_snapshot = {
+        GetAndResetCounter(&percpu_stats.total_successful_requests),
+        // Don't reset total_requests_in_progress because it's
+        // not related to a single reporting interval.
+        percpu_stats.total_requests_in_progress.load(std::memory_order_relaxed),
+        GetAndResetCounter(&percpu_stats.total_error_requests),
+        GetAndResetCounter(&percpu_stats.total_issued_requests),
+        {}};
+    {
+      MutexLock lock(&percpu_stats.backend_metrics_mu);
```

Comment:
> These two lines should go in their own nested scope, so that we're not still holding the lock while we add the per-CPU snapshot to the aggregated snapshot on line 146.

Fixed.